hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/state/initialization/SystemContractsCreator.java

```diff
@@ -0,0 +1,127 @@
+import com.hederahashgraph.api.proto.java.AccountID;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Supplier;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import org.apache.tuweni.bytes.Bytes;
+import org.bouncycastle.util.encoders.Hex;
+
+/**
+ * Encapsulates all the logic for the creation of the enabled system precompiled contracts.
+ *
+ * <p>As of release 0.37.0, there are 3 such contracts:
+ * <ul>
+ *     <li>{@link HTSPrecompiledContract}</li>
+ *     <li>{@link ExchangeRatePrecompiledContract}</li>
+ *     <li>{@link PrngSystemPrecompiledContract}</li>
+ * </ul>
+ */
+@Singleton
+public class SystemContractsCreator {
+
+    public static final Bytes SYSTEM_CONTRACT_BYTECODE =
+            Bytes.concatenate(Bytes.of(Hex.decode("FE00FF")), Bytes.of("SYSTEMCONTRACT".getBytes()));
+    public static final String SYSTEM_CONTRACT_MEMO = "System contract";
+
+    private final Supplier<HederaAccount> accountSupplier;
+    private final BackingStore<AccountID, HederaAccount> accounts;
+    private final List<HederaAccount> contractsCreated = new ArrayList<>();
+    private final PropertySource properties;
+    private final EntityAccess entityAccess;
+
+    @Inject
+    public SystemContractsCreator(
+            final Supplier<HederaAccount> accountSupplier,
+            final BackingStore<AccountID, HederaAccount> accounts,
+            final @CompositeProps PropertySource properties,
+            final EntityAccess entityAccess) {
+        this.accountSupplier = accountSupplier;
+        this.accounts = accounts;
+        this.properties = properties;
+        this.entityAccess = entityAccess;
+    }
+
+    /**
+     * Makes sure that all system contracts, marked as enabled in the {@link PropertyNames#CONTRACTS_SYSTEM_CONTRACTS}
+     * property are present in state, and creates them if necessary.
+     */
+    public void ensureSystemContractsExist() {
+        final var expiry = properties.getLongProperty(BOOTSTRAP_SYSTEM_ENTITY_EXPIRY);
+        final var activeSystemContracts = properties.getAccountNums(CONTRACTS_SYSTEM_CONTRACTS);
+        for (long num = HederaNumbers.FIRST_RESERVED_SYSTEM_CONTRACT;
+                num <= HederaNumbers.LAST_RESERVED_SYSTEM_CONTRACT;
+                num++) {
+            final var id = STATIC_PROPERTIES.scopedAccountWith(num);
+            if (activeSystemContracts.contains(id.getAccountNum())) {
+                if (accounts.contains(id)) {
+                    continue;
+                }
+                final var htsPrecompiledContract = new HederaAccountCustomizer()
+                        .isSmartContract(true)
+                        .isReceiverSigRequired(true)
+                        .isDeclinedReward(true)
+                        .isDeleted(false)
+                        .expiry(expiry)
+                        .memo(SYSTEM_CONTRACT_MEMO)
+                        .key(STANDIN_CONTRACT_ID_KEY)
```

Comment:
Or maybe genesis key 🤔

Reply:
If we are treating them as system accounts, we should use genesis key.